google/cloud/spanner/connection_options_test.cc

```diff
@@ -55,22 +55,14 @@ TEST(ConnectionOptionsTest, AdminEndpoint) {
 }
 
 TEST(ConnectionOptionsTest, SpannerEmulator) {
+  // When SPANNER_EMULATOR_HOST is set, the original endpoint is reset to
+  // ${SPANNER_EMULATOR_HOST}, and the original credentials are reset to
+  // grpc::InsecureChannelCredentials().
   EnvironmentVariableRestore restore("SPANNER_EMULATOR_HOST");
   google::cloud::internal::SetEnv("SPANNER_EMULATOR_HOST", "localhost:9010");
-
   ConnectionOptions options(std::shared_ptr<grpc::ChannelCredentials>{});
-
-  // When SPANNER_EMULATOR_HOST is set, the passed credentials are overridden
-  // (by grpc::InsecureChannelCredentials()) and further updates are ignored.
+  options = internal::EmulatorOverrides(options);
   EXPECT_NE(std::shared_ptr<grpc::ChannelCredentials>{}, options.credentials());
```

Comment:
Are we able to verify that the credentials are indeed the InsecureChannelCredentials rather than just not ChannelCredentials?

Reply:
Two shinny pennies say it is `InsecureChannelCredentials`: that is what we use for Cloud Bigtable, and there is no way we are asking folks to setup trusted SSL certificates to run the emulator.

Reply:
> Are we able to verify that the credentials are indeed the InsecureChannelCredentials rather than just not ChannelCredentials?

Not that I'm aware of.  Note that we are checking that the credentials went from null to non-null, which means `EmulatorOverrides()` did something active.

Reply:
Doh! I misread the question.  I tried checking the type of credentials, could not find a way. They are hidden inside the gRPC "C++ wrappers for C wrappers around C++".